### PR TITLE
Fixed multiple bugs seen in broken texts coming from the Habitus project

### PR DESCRIPTION
**Pull request by MihaiSurdeanu**

### CODE REVIEW

**Comment by MihaiSurdeanu** [comment]
@kwalcock: I believe these bugs are now fixed. If tests pass, can you please merge?
And maybe release 8.4.6?
Thanks!

**Comment by kwalcock** [comment]
In case this wasn't available:

[info] *** 3 TESTS FAILED ***
[error] Failed tests:
[error] 	org.clulab.numeric.TestNumericEntityRecognition

[info] - should recognize literal measurement units *** FAILED ***
[info]   "[B]-MEASUREMENT" was not equal to "[I]-MEASUREMENT" (TestNumericEntityRecognition.scala:382)

[info] - should recognize mass and concentration measurement units *** FAILED ***
[info]   "[B]-MEASUREMENT" was not equal to "[I]-MEASUREMENT" (TestNumericEntityRecognition.scala:382)

[info] - should work correctly with en dashes *** FAILED ***
[info]   "[B]-MEASUREMENT" was not equal to "[I]-MEASUREMENT" (TestNumericEntityRecognition.scala:382)

**Comment by MihaiSurdeanu** [comment]
Thanks @kwalcock ! I pushed something that should fix these, hopefully.

**Comment by kwalcock** [comment]
I tested the new problem document 5995fcf3cf58f14d970c343f.txt locally and the fix seems to work for it as well, so I won't add another unit test.  I'm still looking at the changes here.

File: main/src/main/scala/org/clulab/numeric/NumberParser.scala [hunk 1]
```diff
@@ -43,6 +43,20 @@ object NumberParser {
...
+    for(word <- words if ! seenPlusMinus) {
+      if(word.equals("+/-")) { // TODO: add the Unicode for +/- as well
+        seenPlusMinus = true
+      } else {
+        withoutPlusMinus += word
+      }
+    }
+    withoutPlusMinus
+  }
+
```
Comment (kwalcock) [inline main/src/main/scala/org/clulab/numeric/NumberParser.scala:59]: Is this just
```scala
words.takeWhile(_ != "+/-")
```

**Comment by kwalcock** [comment]
I'm still testing one more small change.

**Comment by kwalcock** [comment]
LGTM now.

**Comment by kwalcock** [comment]
This version ran correctly on all files from habitus that were causing problems.  I will publish it.

**Comment by MihaiSurdeanu** [comment]
+1

Thanks @kwalcock !

**Pull request merged**